swim/memberlist.go

```diff
@@ -283,6 +283,22 @@ func (m *memberlist) Update(changes []Change) (applied []Change) {
 	return applied
 }
 
+// AddJoinList adds the list to the membership with the Update
+// function. However, as a side effect, Update adds changes to
+// the disseminator as well. Since we don't want to disseminate
+// the potentially very large join lists, we clear all the
+// changes from the disseminator, except for the one change
+// that refers to the make-alive of this node.
+func (m *memberlist) AddJoinList(list []Change) {
+	applied := m.Update(list)
+	for _, member := range applied {
+		if member.Address == m.node.Address() {
+			continue
+		}
+		m.node.disseminator.ClearChange(member.Address)
+	}
+}
```

Comment:
The change you are talking about is quite involved. I like it but this PR is focused to just fix the bug. Changing the flow of the code could have some unexpected repercussions.

I think we should keep your suggestion in mind and apply it later though.